src/Catel.Serialization.Json/Catel.Serialization.Json.Shared/Runtime/Serialization/JsonSerialization/JsonSerializer.cs

```diff
@@ -605,10 +613,17 @@ protected override SerializationObject DeserializeMember(ISerializationContext<J
                         var valueType = memberValue.GetBestMemberType();
                         if (valueType.IsEnumEx())
                         {
-                            var enumName = Enum.GetName(valueType, (int)jsonValue);
-                            if (!string.IsNullOrWhiteSpace(enumName))
+                            if (ShouldSerializeUsingEnumToString(memberValue, false))
                             {
-                                finalMemberValue = Enum.Parse(valueType, enumName, false);
+                                finalMemberValue = Enum.Parse(valueType, (string)jsonValue, false);
```

Comment:
We are using a duplicate Enum.Parse here. I think we should store the string value inside a field so we have a single call to Enum.Parse out side the if statements.